src/ifcopenshell-python/ifcopenshell/api/pset/edit_pset.py

```diff
@@ -244,22 +244,46 @@ def add_new_properties(self):
                 continue
             unit, value = self.unpack_unit_value(value)
             if isinstance(value, ifcopenshell.entity_instance):
-                if value.is_a(True) == "IFC4.IfcPropertyEnumeratedValue":
+                if value.is_a("IfcProperty"):
                     properties.append(value)
-                    continue
-                else:
+
+                elif not value.is_entity():
                     args = {"Name": name, "NominalValue": value}
                     if unit:
                         args["Unit"] = unit
                     properties.append(
                         self.file.create_entity("IfcPropertySingleValue", **args)
                     )
+
+                else:
+                    raise ValueError(f"'{value}' is not a valid property value for '{name}'")
```

Comment:
Maybe include .is_a() for this edge case TypeError? -> `f"A {value.is_a()} cannot be assigned to the property set {name}"`

Reply:
Sure, but I believe `TypeError` is used for wrong Python type used, in this case the type is valid - `entity_instance`, but its ifc class is not. So I lean towards `ValueError` more.